diff --git a/src/ML-METATOMIC/metatomic_system.cpp b/src/ML-METATOMIC/metatomic_system.cpp
--- a/src/ML-METATOMIC/metatomic_system.cpp
+++ b/src/ML-METATOMIC/metatomic_system.cpp
@@ -699,7 +699,7 @@ void MetatomicSystemAdaptor::add_masses(metatomic_torch::System& system, double
     masses = masses * unit_conversion;
     auto tensor = make_per_atom_tensormap(masses, dtype, device, "mass");
 
-    system->add_data("masses", tensor, /*override=*/true);
+    system->add_data("masses", tensor, /*override=*/true, /*private_warn_on_deprecated=*/true);
 }
 
 
@@ -734,7 +734,7 @@ void MetatomicSystemAdaptor::add_momenta(metatomic_torch::System& system, double
     momenta = momenta * unit_conversion;
     auto tensor = make_per_atom_tensormap(momenta, dtype, device, "momentum", {"xyz"});
 
-    system->add_data("momenta", tensor, /*override=*/true);
+    system->add_data("momenta", tensor, /*override=*/true, /*private_warn_on_deprecated=*/true);
 }
 
 void MetatomicSystemAdaptor::add_velocities(metatomic_torch::System& system, double unit_conversion) {
@@ -764,5 +764,5 @@ void MetatomicSystemAdaptor::add_velocities(metatomic_torch::System& system, dou
     velocities = velocities * unit_conversion;
     auto tensor = make_per_atom_tensormap(velocities, dtype, device, "velocity", {"xyz"});
 
-    system->add_data("velocities", tensor, /*override=*/true);
+    system->add_data("velocities", tensor, /*override=*/true, /*private_warn_on_deprecated=*/true);
 }
